### PR TITLE
Split up Painless Fields API conversion tests

### PR DESCRIPTION
**Pull request by jdconrad**
This change has three small parts.
* Refactors the tests to have a base test class that contains predefined fields for each relevant type conversion. This base test class allows availability of predefined fields for any new fields added outside of core.
* Splits the conversions test cases into a one to one matching for each possible conversion where we do field type to other field type, and if possible, the reverse as well within a single test.
* Fixes a bug where we relied on the delegating method values for get long and get double, but in some cases this was giving non-converted data.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/script/field/LongField.java [hunk 1]
```diff
@@ -82,11 +82,27 @@ public static BigIntegerField toBigIntegerField(LongField sourceField) {
 
             @Override
             public BigInteger getNonPrimitiveValue() {
-                return BigInteger.valueOf(values.getLongValue());
+                return toBigInteger(values.getNonPrimitiveValue());
+            }
+
+            @Override
+            public long getLongValue() {
+                return toBigInteger(values.getNonPrimitiveValue()).longValue();
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/script/field/LongField.java:90]: Why convert from a long, to a BigInteger, back to long?

File: server/src/test/java/org/elasticsearch/script/field/ConvertersTestBase.java [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+            }
+
+            @Override
+            public int size() {
+                return rawBooleanValues.length;
+            }
+
+            @Override
+            public List<Boolean> getValues() {
+                List<Boolean> values = new ArrayList<>();
```
Comment (stu-elastic) [inline server/src/test/java/org/elasticsearch/script/field/ConvertersTestBase.java:58]: Could `rawBooleanValues` change? If not, why not `Arrays.asList(rawBooleanValues)`?

**Review by stu-elastic — APPROVED** [review]
A couple of questions

File: server/src/test/java/org/elasticsearch/script/field/ConvertersTestBase.java (hunk 2 above)
Comment (jdconrad) [reply server/src/test/java/org/elasticsearch/script/field/ConvertersTestBase.java:58]: Apparently, this is unsupported. I poked around for a while trying to get the correct stream operation.

File: server/src/main/java/org/elasticsearch/script/field/LongField.java (hunk 1 above)
Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/LongField.java:90]: Going through all of the converters and removing unnecessary additional conversions.

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review. Updated the code based on your feedback. Will commit as soon as CI passes.

**Pull request merged**